collect_app/src/main/java/org/odk/collect/android/widgets/TimeWidget.java

```diff
@@ -15,295 +15,106 @@
 package org.odk.collect.android.widgets;
 
 import android.annotation.SuppressLint;
-import android.app.TimePickerDialog;
+import android.app.Activity;
 import android.content.Context;
-import android.content.res.TypedArray;
-import android.graphics.Color;
-import android.graphics.drawable.ColorDrawable;
-import android.os.Build;
-import android.text.format.DateFormat;
-import android.util.AttributeSet;
-import android.view.Window;
-import android.widget.Button;
-import android.widget.LinearLayout;
-import android.widget.TextView;
-import android.widget.TimePicker;
+import android.util.TypedValue;
+import android.view.View;
 
 import org.javarosa.core.model.data.IAnswerData;
-import org.javarosa.core.model.data.TimeData;
+import org.javarosa.form.api.FormEntryPrompt;
 import org.joda.time.DateTime;
 import org.odk.collect.android.R;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.databinding.WidgetAnswerBinding;
 import org.odk.collect.android.formentry.questions.QuestionDetails;
-import org.odk.collect.android.formentry.questions.WidgetViewUtils;
-import org.odk.collect.android.widgets.interfaces.ButtonClickListener;
+import org.odk.collect.android.widgets.interfaces.BinaryDataReceiver;
+import org.odk.collect.android.widgets.utilities.DateTimeWidgetUtils;
 
-import java.lang.reflect.Constructor;
-import java.lang.reflect.Field;
 import java.util.Date;
 
-import timber.log.Timber;
-
-import static org.odk.collect.android.formentry.questions.WidgetViewUtils.createAnswerTextView;
-import static org.odk.collect.android.formentry.questions.WidgetViewUtils.createSimpleButton;
-
-/**
- * Displays a TimePicker widget.
- *
- * @author Carl Hartung (carlhartung@gmail.com)
- */
 @SuppressLint("ViewConstructor")
-public class TimeWidget extends QuestionWidget implements ButtonClickListener, TimePickerDialog.OnTimeSetListener {
-    private TimePickerDialog timePickerDialog;
-
-    Button timeButton;
-    final TextView timeTextView;
+public class TimeWidget extends QuestionWidget implements BinaryDataReceiver {
+    WidgetAnswerBinding binding;
 
     private int hourOfDay;
```

Comment:
I think it's not consistent than in this widget we keep separate values hour, minute and in DateWidget we keep LocalDateTime wouldn't it be better to have TimeData field here instead of hour/minute?